crates/holochain_sqlite/build.rs

```diff
@@ -68,9 +75,44 @@ fn check_fmt(path: &std::path::Path) {
     }
 }
 
+fn check_migrations() {
+    use std::process::Command;
+    let root = PathBuf::from(SQL_DIR);
+    for dir in [
+        root.join("cell/schema"),
+        root.join("conductor/schema"),
+        root.join("p2p_agent_store/schema"),
+        root.join("p2p_metrics/schema"),
+        root.join("wasm/schema"),
+    ] {
+        for path in find_sql(&dir) {
+            let mut cmd = Command::new("git");
+            match cmd
+                .arg("diff")
+                // Filter out files which were merely added
+                .arg("--diff-filter=a")
+                .arg(path.clone())
+                .output()
+            {
+                Ok(out) => {
+                    if out.status.success() && out.stdout.is_empty() && out.stderr.is_empty() {
+                        // no change. good.
+                    } else {
+                        panic!("Diff found in schema file:\n\n{}\nSchema and migration files cannot be modified. Instead, set up a new database migration in 'crates/holochain_sqlite/src/schema.rs'\n\n", String::from_utf8_lossy(&out.stdout))
```

Comment:
I'm not saying no to this... it's just a little oddly disconnected, like, if the dev happens to commit a checkpoint without running a build, then when they do run the build this won't error, right?

Reply:
Yeah I was trying to find a more robust check, that also doesn't get in the way, can you think of one?

Reply:
I guess, at the very least you could `git diff origin/develop`, though technically folks could rename their origin ref to something else and would cause that to break...

Reply:
@neonphog yeah I don't know if we can have a failsafe check which is set-and-forget, since the goalpost of "has not changed" is always moving. Whatever we can do to help remind ourselves will be helpful but we ultimately have to remember what this is all for.

Reply:
Let's try `origin/develop`